components/settings/index.js

```diff
@@ -157,15 +157,26 @@ function Settings (opts) {
 /**
  *
  */
-Settings.prototype.getDataUrl = function () {
-  // TODO pass in zoom level
+Settings.prototype.getDataUrl = function (zoom) {
+  console.info('getting data url for zoom level', zoom)
+  var fallback
   var backends = this.tabData.storage.backends
+
   for (var i = 0; i < backends.length; ++i) {
     var data = backends[i]
     if (typeof data.url === 'string' && data.active) {
-      return data.url
+      if (!fallback) fallback = data
+      if (data.zoom[0] <= zoom && data.zoom[1] >= zoom) {
+        return data.url
+      }
     }
   }
+
+  if (fallback) {
+    return fallback.url
+  } else {
+    console.warn('no matching data url for zoom level', zoom)
```

Comment:
Leaving for debug purposes to see what's going on when you zoom out and in etc. Can clean up later.